iceoryx_examples/automotive_soa/include/owl/kom/field_subscriber.inl

```diff
@@ -0,0 +1,182 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_FIELD_SUBSCRIBER_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_FIELD_SUBSCRIBER_INL
+
+#include "owl/kom/field_subscriber.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline FieldSubscriber<T>::FieldSubscriber(const ServiceIdentifier& service,
+                                           const InstanceIdentifier& instance,
+                                           const FieldIdentifier& field) noexcept
+    : m_subscriber({service, instance, field}, {QUEUE_CAPACITY, HISTORY_REQUEST})
+    , m_client({service, instance, field})
+{
+    m_waitset.attachState(m_client, iox::popo::ClientState::HAS_RESPONSE).expect("Failed to attach client!");
+}
+
+template <typename T>
+inline FieldSubscriber<T>::~FieldSubscriber() noexcept
+{
+    m_waitset.detachState(m_client, iox::popo::ClientState::HAS_RESPONSE);
+
+    // Wait here if a thread or callback is still running
+    while (m_threadsRunning > 0)
+    {
+    }
+    std::lock_guard<iox::posix::mutex> guard(m_mutex);
+}
+
+template <typename T>
+template <typename Callable>
+inline core::Result<size_t> FieldSubscriber<T>::TakeNewSamples(Callable&& callable, size_t maxNumberOfSamples) noexcept
+{
+    core::Result<size_t> numberOfSamples{0};
+
+    // Hint: Depending on the type of callable, it needs to be checked for null or restricting to lambdas
+
+    while (m_subscriber.take()
+               .and_then([&](const auto& sample) {
+                   callable(sample.get());
+                   numberOfSamples++;
+               })
+               .or_else([](auto& result) {
+                   if (result != iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)
+                   {
+                       std::cerr << "Error receiving chunk!" << std::endl;
+                   }
+               })
+           && numberOfSamples < maxNumberOfSamples)
+    {
+    }
+    return numberOfSamples;
+}
+
+template <typename T>
+inline Future<T> FieldSubscriber<T>::Get()
+{
+    // If we call Get() twice shortly after each other, once the response of the first request has not yet
+    // arrived, we have a problem
+    //! [FieldSubscriber get]
+    bool requestSuccessfullySent{false};
+    m_client.loan()
+        .and_then([&](auto& request) {
+            request.getRequestHeader().setSequenceId(m_sequenceId);
+            request.send().and_then([&]() { requestSuccessfullySent = true; }).or_else([](auto& error) {
+                std::cerr << "Could not send Request! Error: " << error << std::endl;
+            });
+        })
+        .or_else([&](auto& error) {
+            std::cerr << "Could not allocate Request! Error: " << error << std::endl;
+            requestSuccessfullySent = false;
```

Comment:
Already set to false. I read the algorithm with the beginning initialization of `requestSuccessfullySent{false}` like: it fails until someone states that it succeeded. But when you set it here to false I start to wonder if it is possible that someone sets it to true and I overlooked something.

But this is just some personal opinion.